include/amici_solver.h

```diff
@@ -189,6 +231,326 @@ class Solver {
       */
     virtual void turnOffRootFinding() = 0;
 
+    /** sensitivity method
+     * @return method enum
+     */
+    AMICI_sensi_meth getSensitivityMethod() const{
+        return sensi_meth;
+    }
+
+    /**
+     * @brief setSensitivityMethod
+     * @param sensi_meth
+     */
+    void setSensitivityMethod(AMICI_sensi_meth sensi_meth) {
+        this->sensi_meth = sensi_meth;
+    }
+
+    /**
+     * @brief getNewtonMaxSteps
+     * @return
+     */
+    int getNewtonMaxSteps() const {
+        return newton_maxsteps;
+    }
+
+    /**
+     * @brief setNewtonMaxSteps
+     * @param newton_maxsteps
+     */
+    void setNewtonMaxSteps(int newton_maxsteps) {
+        this->newton_maxsteps = newton_maxsteps;
+    }
+
+    /**
+     * @brief getNewtonPreequilibration
+     * @return
+     */
+    bool getNewtonPreequilibration() const {
+        return newton_preeq;
+    }
+
+    /**
+     * @brief setNewtonPreequilibration
+     * @param newton_preeq
+     */
+    void setNewtonPreequilibration(bool newton_preeq) {
+        this->newton_preeq = newton_preeq;
+    }
+
+    /**
+     * @brief getNewtonPreconditioner
+     * @return
+     */
+    int getNewtonPreconditioner() const {
+        return newton_precon;
+    }
+
+    /**
+     * @brief setNewtonPreconditioner
+     * @param newton_precon
+     */
+    // TODO: what are valid options?
```

Comment:
I believe this is currently unused and will be removed (see #221) @paulstapor ?

Reply:
The preconditioning is currently done in a fixed way. I implemented an option for this, since Glenn wanted to test different preconditioning schemes at some point... However, this is not going to happen within the next few weeks, so I would rather remove it...

Reply:
Since it involves changing quite some files back and forth, I would leave it there if it is likely to be implemented during this year.